### PR TITLE
Adding Login and Signup Screen

### LINKED ISSUE #82: Design Login and Signup Screen For the app
Login and Signup Feature  is added in android app . It can be added in iOS app too. Design the screen of Login and Signup for iOS app.

Would you like to work on this issue?
Yes

### LINKED ISSUE #84: Add Signup Functionality
Signup screen needs to be designed and implemented.Signup feature is implemented in android.

Would you like to work on this issue?
Yes

### PR DESCRIPTION
**Pull request by SIMRAN1**
Fixes  #82
Login and Signup Screens are added.

### CODE REVIEW

**Comment by SIMRAN1** [comment]
@imjog @chashmeetsingh 
image
image
These are screens of Android app should I design similar to it?

**Comment by jogendra** [comment]
@SIMRAN1 Please post the screenshot of changes you made.

**Comment by SIMRAN1** [comment]
@imjog I have to design in iOS app login and sign up screen that is why I discussed should I make similar to Android screen?

**Comment by jogendra** [comment]
@SIMRAN1 then what changes you made in this PR? The description is saying _Login and Signup Screens are added._

**Comment by SIMRAN1** [comment]
@imjog for now I have added only TabBarViewController Login and signup screens are blank . I will be updating PR in a while.

**Comment by jogendra** [comment]
@SIMRAN1 Add [WIP] to the PR till the work in progress.

**Comment by mooocer** [comment]
@SIMRAN1 If it helps, this is how the Login, SignUp and Forgot Password screens look in the Organizer Android App.

videotogif_2018 05 26_21 02 30

**Comment by SIMRAN1** [comment]
@imjog @chashmeetsingh Here are the screens designed for iOS . please review
image
image

**Review by jogendra — CHANGES_REQUESTED** [review]
- UI is not similar to android
- TabBar icons are missing
- Profile tab icon is missing
- The keyboard is not working the way it should work (on all text fields)
- Navigation Bar on Login and Signup screen do not have title and color as in Android
- Login and Signup button looks odd to me
- Background colors on login and signup screen is not matching with Android
- There is no way to go to back in the app from login and Signup screen, this is a very bad thing a user can experience
- Status bar color is not as expected
- On main tab bar, the **more** tab always should be last tab, which is not currently
- What will happed when you press last or second last text field, the button and textfield hidden behind keyboard, this is very bad practice so use ScrollView and scroll accordingly use input in textfield
- Corresponding classes  files are not created to storyboard view controllers

This PR need a lot of improvements. Please check by youself before making PR that things are as expected.

**Comment by SIMRAN1** [comment]
@imjog  I am doing the changes .Pushing commits in a while!!.Actually I had only designed screens in story board previously. Now I am adding other functionalities and its controllers!!

**Comment by SIMRAN1** [comment]
image
image
image

**Comment by SIMRAN1** [comment]
@imjog I have made almost all the changes . Screens are little bit different from Android  as There theme color is different . @imjog @chashmeetsingh @jurvis please review!

File: FOSSAsia.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -7,6 +7,13 @@
 	objects = {
 
 /* Begin PBXBuildFile section */
+		1884C8DC20D639F8004B221F /* SignupVcMethods.swift in Sources */ = {isa = PBXBuildFile; fileRef = 1884C8DB20D639F8004B221F /* SignupVcMethods.swift */; };
+		1884C8DF20D642E6004B221F /* LoginViewController.swift in Sources */ = {isa = PBXBuildFile; fileRef = 1884C8DE20D642E6004B221F /* LoginViewController.swift */; };
+		1884C8E120D6430B004B221F /* LoginVCMethods.swift in Sources */ = {isa = PBXBuildFile; fileRef = 1884C8E020D6430B004B221F /* LoginVCMethods.swift */; };
+		18D828B120D68A0600159EC0 /* Bordered Button.swift in Sources */ = {isa = PBXBuildFile; fileRef = 18D828B020D68A0600159EC0 /* Bordered Button.swift */; };
```
Comment (jogendra) [inline FOSSAsia.xcodeproj/project.pbxproj]: Remove it!

**Review by jogendra — CHANGES_REQUESTED** [review]
- Give back icon just before **Back** on Login and Signup navbar
- Login and Signup button is not as expected - Give them red-type (the theme color of app) as background and title color to white, give small corner radius as well.
- Signup screen should be in `scrollView`, again what happens when user want to input in last text field?
- Keyboard is not handled on `textFields`, current pressing on keyboard crashing the app, taping on screen and pressing `return` should dismiss the keyboard.

File: FOSSAsia.xcodeproj/project.pbxproj (hunk 1 above)
Comment (SIMRAN1) [reply FOSSAsia.xcodeproj/project.pbxproj]: okk @imjog !! but Signup screen is in Scroll View only!!

Comment (jogendra) [reply FOSSAsia.xcodeproj/project.pbxproj]: I can't able to scroll it

Comment (SIMRAN1) [reply FOSSAsia.xcodeproj/project.pbxproj]: Okk @imjog I will see to it then!

**Comment by SIMRAN1** [comment]
@imjog I have done the changes .Its working fine in simulator but I dont have real device to test it. Please Review and minor fixing I will do in another Pr in which I have to add functionality of login and Signup.

**Review by jogendra — CHANGES_REQUESTED** [review]
- Put signup screen in `scrollView`
- Handle `Return` key in keyboards
- Increase Login and Signup buttons heigh to 44
- Look into falling Codacy

**Comment by jogendra** [comment]
@SIMRAN1 You might face the issue while putting the Signup screen in `scrollView`. For this, you can refer SUSI iOS `SkillDetailViewController` where I used `scrollView` or there are some good answers on stackoverflow you can refer. Another solution is, delete current element from Signup screen and start with new `ScrollView`. If you get stucked, rather than wasting time, you can always ask help from us. Good work with other implementation. thanks

**Comment by SIMRAN1** [comment]
@imjog Signup Screen is already in ScrollView and is working perfectly fine in simulator . I am not able to get why it is not working in real device . Will figure it out soon .

**Comment by SIMRAN1** [comment]
@imjog @chashmeetsingh @mariobehling I request you to merge this PR as it is acting as blocker for other PRs . @imjog for Scroll View I will create another PR and fix it but currently this PR needs to be merged thanks.

**Comment by SIMRAN1** [comment]
@imjog I have solved this problem of ScrollView  in another  PR Adding signup Functionality.

**Review by jogendra — APPROVED** [review]
Look fine to me. Please ask review from Android and web team as well.

**Comment by SIMRAN1** [comment]
@simarsingh24 @nikit19 @dreadpool2 please review pr!!

**Review by dreadpool2 — APPROVED** [review]
@SIMRAN1 Looks great :+1:

**Comment by SIMRAN1** [comment]
@mariobehling please review!

**Review by chashmeetsingh — APPROVED** [review]

**Pull request merged**